include/pybind11/eigen.h

```diff
@@ -265,14 +316,47 @@ struct type_caster<Type, enable_if_t<is_eigen_dense_plain<Type>::value>> {
         auto fits = props::conformable(buf);
         if (!fits)
             return false;
-
+        int result = 0;
         // Allocate the new type, then build a numpy reference into it
         value = Type(fits.rows, fits.cols);
-        auto ref = reinterpret_steal<array>(eigen_ref_array<props>(value));
-        if (dims == 1) ref = ref.squeeze();
-        else if (ref.ndim() == 1) buf = buf.squeeze();
 
-        int result = detail::npy_api::get().PyArray_CopyInto_(ref.ptr(), buf.ptr());
+        if (npy_format_descriptor<Scalar>::value == npy_api::NPY_OBJECT_) {
```

Comment:
BTW For consistency, could you place this value into `is_py_object`? And same as above, could this be transposed / reversed to keep the original code more visible?

Reply:
Done.